### DISCUSSION TITLE
Keyboard-Shortcut to Swap Joystick port

### DISCUSSION THREAD

**Discussion by sgrockfort** [vossstef/tang_nano_20k_c64#88] [Ideas]
Hello,

I'd love to have a keyboard shortcut which swap the joystick ports :-) Maybe an idea for a future release?!

cheers 
Michael

**Comment by geniac** [comment]
+1
I'd also love to see this feature!
Would it be possible to implement this function on the "S1 Button"?

Thanks for this Amazing Project!

Greetings

Lars

**Reply by vossstef** [reply]
let me have a look into that over the next weeks.

**Reply by vossstef** [reply]
@geniac @sgrockfort 
i added Joystick swap support in the latest release. Would you pls. check and give Feedback? thx

**Reply by geniac** [reply]
@vossstef 

Just updated, and tested.

Swapping controls works like a charm now :)

I've noticed that when using DS2 the second Trigger button is recieved by port#1 & port#2 simultaneously.
I am not 100% sure if this is the expected behaviour.

Edit: tested again and i am unable to reproduce this behaviour

Greetings

Lars

**Reply by vossstef** [reply]
@geniac 
thx !
will check the issue.
Got already feedback that the 2nd trigger is also buggy so there is work ahead.

**Comment by sgrockfort** [comment]
Hi,

I just tested 1.5.1 and it works just perfect :-) 
@vossstef thank you so much !for you work !!!!

cheers 
Michael

**Reply by vossstef** [reply]
Thanks!🙃
